packages/cli-e2e/__tests__/vue.specs.ts

```diff
@@ -0,0 +1,111 @@
+import retry from 'async-retry';
+import {spawn, spawnSync} from 'child_process';
+
+import type {ChildProcessWithoutNullStreams} from 'child_process';
+
+import {CLI_EXEC_PATH, killCliProcessFamily} from '../utils/cli';
+import {closeAllPages, getBrowser} from '../utils/browser';
+import {Browser, HTTPRequest, Page} from 'puppeteer-core';
+import {loginWithOffice} from '../utils/login';
+import {isSearchRequest} from '../utils/platform';
+
+describe('ui', () => {
+  describe('create:vue', () => {
+    let browser: Browser;
+    const cliProcesses: ChildProcessWithoutNullStreams[] = [];
+    // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
+    const clientPort = '8080';
+    const projectName = 'vue-project';
+    const searchPageEndpoint = `http://localhost:${clientPort}`;
+    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
+    let interceptedRequests: HTTPRequest[] = [];
+    let page: Page;
+    let waitForProjectToStartTimeout: NodeJS.Timeout;
+
+    beforeAll(async () => {
```

Comment:
I think there's some thing we could extract in a shared file, similar to what you have for `loginWithOffice`.

Let's say a `createUIProject()` function, that receives the CLI command to execute ( in this case: `ui:create:vue`) , and a command to run when the project is generated ( in this case npm run start ).

It takes care of doing all the boilerplate here, as well as the not so obvious `detached process` configuration.

I can definitely see this being reused for all UI commands, at least.